File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.py
class Solution:
    def check(self, nums: List[int]) -> bool:
        n = len(nums)
        peak = 0

        for i in range(n):
            # (i + 1) % n seamlessly wraps the last element back to the first element
            if nums[i] > nums[(i + 1) % n]:
                peak += 1

        # A valid sorted and rotated array can have at most 1 drop/peak
        return peak <= 1

File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.js
var check = function (nums) {
    const n = nums.length
    const sortedArr = [...nums].sort((a, b) => a - b)

    for (let r = 0; r < n; r++) {
        isRotated = true
        for (let i = 0; i < n; i++) {
            if (sortedArr[i] !== nums[(i + r) % n]) {
                isRotated = false
                break
            }
        }
        if (isRotated) {
            return true
        }
    }
    return false
};